### DISCUSSION TITLE
How to convert 2D vector to 3D vector in GeoPlotter

### DISCUSSION THREAD

**Discussion by tkoyama010** [bjlittle/geovista#384] [Ideas]
When plotting the following vectors drawn in cartopy in GeoVista, the 2D vectors need to be converted to vectors of 3D coordinates. This thread discusses this mathematical aspect and prepares for future implementation in GeoVista.
![arrows_00_001](https://github.com/bjlittle/geovista/assets/7513610/35acb866-7d19-4564-842f-ae96bc1ff6db)
The reference document is below. (The language of the document is Japanese for the convenience of the author. Please translate it and check it.)
https://physmemo.shakunage.net/spherical/spherical.html

**Comment by bjlittle** [comment]
@dennissergeev Do you have prior experience of this? I know that you've dabbled in this space :thinking:

**Comment by dennissergeev** [comment]
Thanks for raising this @tkoyama010, this is something I really want to see in `geovista` too! (see #16)

A while ago I added [this function](https://github.com/pyvista/pyvista/blob/aee835c5a79cb9de945c8e2971fabc9096b0feb1/pyvista/core/utilities/features.py#L150) to `pyvista`. A somewhat convoluted example of applying it to some model data is [here](https://github.com/exoclim/aeolus/blob/main/src/aeolus/plot/pv.py#L131).

It'd be great to create a clean and convenient function to plot vector glyphs in `geovista` using these building blocks...
